### ISSUE TITLE
'If-None-Match' header causes the API to reject consecutive calls

### ISSUE THREAD

**Issue by daninfpj** [cakejelly/layer-api#4]
There's a problem with the way you're handling the 'If-None-Match' header. The token is created when you instantiate the platform object. So if you use that instance to say create two messages, layer will reject the second one:

```
conv = platform.conversations.find("conversation_id")
conv.messages.create(message)
conv.messages.create(another_message)
```

`another_message` will get rejected and the message won't be send since the API already saw that token and will interpret it as a duplicate.

**Comment by cakejelly** [comment]
Yikes! Thanks for reporting, will get this sorted ASAP

**Comment by cakejelly** [comment]
I took a look into this and can't seem to reproduce anything. [HttpClient#default_layer_headers](https://github.com/cakejelly/layer-api/blob/master/lib/layer/http_client.rb#L53) is called every time a HTTP request is sent, which will generate a fresh token each time.

Have you experienced layer rejecting any of your requests from using this gem?

**Comment by cakejelly** [comment]
Apologies for the delay here.. finally managed to reproduce & have pushed a fix in #7 . Thanks again for helping out.
